src/main/java/com/github/manolo8/darkbot/backpage/DispatchManager.java

```diff
@@ -0,0 +1,172 @@
+        String page = manager.getConnection("indexInternal.es?action=internalDispatch", Method.GET, 10_000).getContent();
+
+        if (page == null || page.isEmpty()) return false;
+        lastDispatcherUpdate = System.currentTimeMillis();
+        return InfoReader.updateAll(page, data);
+    }
+
+    private enum InfoReader {
+        PERMIT("name=\"permit\" value=\"([0-9]+)\"", DispatchData::setPermit),
+        GATE_UNIT("name=\"permit\" value=\"([0-9]+)\"", DispatchData::setGateUnits),
+        SLOTS(":([0-9]+).*class=\"userCurrentMax\">([0-9]+)", DispatchData::setAvailableSlots, DispatchData::setMaxSlots),
+        ITEMS("<tr class=\"dispatchItemRow([\\S\\s]+?)</tr>", DispatchData::parseRow);
+
+        private final Pattern regex;
+        private final List<BiConsumer<DispatchData, String>> consumers;
+
+        InfoReader(@Language("RegExp") String regex, List<BiConsumer<DispatchData, String>> consumers) {
+            this.regex = Pattern.compile(regex);
+            this.consumers = consumers;
+        }
+
+        @SafeVarargs
+        InfoReader(@Language("RegExp") String regex, BiConsumer<DispatchData, String>... consumers) {
+            this(regex, Arrays.asList(consumers));
+        }
+
+        @SafeVarargs
+        InfoReader(@Language("RegExp") String regex, BiIntConsumer<DispatchData>... consumers) {
+            this(regex, Arrays.stream(consumers)
+                    .map(c -> (BiConsumer<DispatchData,String>) (d, s) -> c.accept(d, Integer.parseInt(s)))
+                    .collect(Collectors.toList()));
+        }
+
+        private boolean update(String page, DispatchData data) {
+            Matcher m = regex.matcher(page);
+            if (!m.find()) return false;
+
+            do {
+                int max = Math.min(m.groupCount(), consumers.size());
+                for (int i = 0; i < max; i++)
+                    consumers.get(i).accept(data, m.group(i + 1));
+            } while (m.find());
+
+            return m.groupCount() == consumers.size();
+        }
+
+        public static boolean updateAll(String page, DispatchData data) {
+            boolean updated = true;
+            for (InfoReader reader : InfoReader.values()) {
+                updated &= reader.update(page, data);
+            }
+            return updated;
+        }
+    }
+
+    public boolean hireRetriever(Retriever retriever){
+        if(data.getAvailableSlots() > 0) {
+            //retriever.getCost(); // TODO: check cost
+            try {
+                String response = main.backpage.getConnection("ajax/dispatch.php", Method.POST)
+                        .setRawParam("command", "sendDispatch")
+                        .setRawParam("dispatchId", retriever.getId())
+                        .getContent();
+                if(response.contains("\"result\":\"ERROR\"")){
+                    System.out.println("Failed to dispatch " + retriever.getId() + ": " + response);
+                    update(10_000);
```

Comment:
This 10_000 will make the update get ignored most of the time, since if you first update, and then hire (the usual flow if you want up-to-date data) it will not have been 10 seconds betweeen
```suggestion
                    update(-1);
```